### PR TITLE
Made it possible to use IEnumerable.Empty() in HasDefaultValue.

### LINKED ISSUE #17780: HasDefaultvalue() for interfaces/generics
Unable to set default value for interfaces/generics.

Not working:

Model:

```C#
public class EventTriggers
{
    public IEnumerable<Regex> Triggers { get; private set; }
}
```

Context Configuration:
```C#
builder.OwnsOne(t => t.EventTriggers, trigger =>
    {
        trigger.Property(p => p.Triggers)
            .HasConversion<string>(
                to => to.SerializeToJson(Newtonsoft.Json.Formatting.None, true),
                from => from.DeserializeFromJson<IEnumerable<Regex>>())
            .HasDefaultValue(Enumerable.Empty<Regex>());          
    });
```

Results in the following when running add-migration:

```
Cannot set default value 'System.Text.RegularExpressions.Regex[]' of type 'System.Text.RegularExpressions.Regex[]' on property 'Triggers' of type 'System.Collections.Generic.IEnumerable`1[System.Text.RegularExpressions.Regex]' in entity type 'EventTriggers'.
```

The following is working:

Model:

```C#
public class EventTriggers
{
   public List<Regex> Triggers { get; private set; }
}
```

Context Configuration:

```C#
builder.OwnsOne(t => t.EventTriggers, trigger =>
    {
        trigger.Property(p => p.Triggers)
            .HasConversion<string>(
                to => to.SerializeToJson(Newtonsoft.Json.Formatting.None, true),
                from => from.DeserializeFromJson<List<Regex>>())
            .HasDefaultValue(new List<Regex>());          
    });
```

Is this by design?

EF Core version: 2.2.6
Database provider: npgsql
Target framework: (e.g. .NET Core 2.2)
Operating system: Cross Platform
IDE: VS 2019

### PR DESCRIPTION
**Pull request by TobiasNissen**
Made it possible to use IEnumerable.Empty() in HasDefaultValue.
 - Also added a test for it.

Fixes #17780

### CODE REVIEW

**Comment by TobiasNissen** [comment]
Please let me know if the bug should be fixed differently.

File: src/EFCore.Relational/Extensions/RelationalPropertyExtensions.cs [hunk 1]
```diff
@@ -334,7 +335,14 @@ private static object ConvertDefaultValue([NotNull] IProperty property, [CanBeNu
             {
                 try
                 {
-                    return Convert.ChangeType(value, property.ClrType, CultureInfo.InvariantCulture);
+                    if (typeof(IEnumerable).IsAssignableFrom(property.ClrType) && property.ClrType.IsGenericType)
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Extensions/RelationalPropertyExtensions.cs]:
```suggestion
                    if (typeof(IEnumerable).IsAssignableFrom(property.ClrType)
                        && property.ClrType.IsGenericType
                        && property.ClrType.IsAssignableFrom(valueType))
```

File: test/EFCore.Relational.Tests/Metadata/RelationalBuilderExtensionsTest.cs [hunk 2]
```diff
@@ -1205,6 +1205,16 @@ public void Relational_relationship_methods_have_non_generic_overloads()
...
 
+        [ConditionalFact]
+        public void Can_use_empty_IEnumerable_as_default_value()
+        {
+            var modelBuilder = CreateConventionModelBuilder();
+            modelBuilder
+                .Entity<Customer>()
+                .Property(c => c.Orders)
+                .HasDefaultValue(Enumerable.Empty<Order>());
+        }
```
Comment (AndriySvyryd) [inline test/EFCore.Relational.Tests/Metadata/RelationalBuilderExtensionsTest.cs]: Also add an assert to check the default value was set

**Comment by ajcvickers** [comment]
Closing this for now since there has been no recent activity. To resurrect this for more work, please create a new PR re-based on current master.

**Pull request closed without merging**